app/presenters/tasks/case_details.rb

```diff
@@ -20,7 +20,7 @@ def in_progress?
     def completed?
       crime_application.case.values_at(
         :hearing_court_name, :hearing_date
-      ).all?(&:present?)
+      ).all?(&:present?) && crime_application.case.valid?
```

Comment:
Could the new validation `crime_application.case.valid` be added as a guard clause to make it easier to follow the rule?

Reply:
@mseedat-moj I removed this additional validation for time being to support old applications.  For new applications Case model is validating the  object before saving. 
I think definition of completed? needs to be revised